src/scheduling.rs

```diff
@@ -118,21 +125,29 @@ impl Worker {
         config_path: String,
     ) -> Worker {
         let id_copy = id.clone();
-        let (sender, receiver) = mpsc::channel();
+        let (sender, receiver) = mpsc::channel(1);
         let thread = match worker_type {
             // TODO: Spawn a new worker thread when one dies? (A supervisor or something like that?)
             WorkerType::Alert => thread::spawn(move || {
-                if let Err(e) =
-                    alert_worker::alert_worker(&id, stream_name.clone(), &config_path, receiver)
-                {
-                    error!(id, stream_name, config_path, error = %e, "alert worker failed")
+                if stream_name == "ZTF" {
+                    let _ = run_alert_worker::<ZtfAlertWorker>(id, receiver, &config_path);
+                } else if stream_name == "LSST" {
+                    let _ = run_alert_worker::<LsstAlertWorker>(id, receiver, &config_path);
+                } else {
+                    panic!("Unknown stream name: {}", stream_name);
```

Comment:
Using `let _ = ...` effectively discards any error that occurs. How about this instead?

```rust
            WorkerType::Alert => thread::spawn(move || {
                let run = match stream_name.as_str() {
                    "ZTF" => run_alert_worker::<ZtfAlertWorker>,
                    "LSST" => run_alert_worker::<LsstAlertWorker>,
                    _ => {
                        error!("Unknown stream name: {}", stream_name);
                        return;
                    }
                };
                if let Err(error) = run(id, receiver, &config_path) {
                    error!(error = %error, "failed to run alert worker");
                }
            }),
```

Reply:
done